executor/ddl.go

```diff
@@ -438,9 +519,20 @@ func (e *DDLExec) dropTableObject(objects []*ast.TableName, obt objectType, ifEx
 	if len(notExistTables) > 0 && ifExists {
 		for _, table := range notExistTables {
 			if obt == sequenceObject {
-				e.ctx.GetSessionVars().StmtCtx.AppendNote(infoschema.ErrSequenceDropExists.GenWithStackByArgs(table))
+				sessVars.StmtCtx.AppendNote(infoschema.ErrSequenceDropExists.GenWithStackByArgs(table))
 			} else {
-				e.ctx.GetSessionVars().StmtCtx.AppendNote(infoschema.ErrTableDropExists.GenWithStackByArgs(table))
+				sessVars.StmtCtx.AppendNote(infoschema.ErrTableDropExists.GenWithStackByArgs(table))
+			}
+		}
+	}
+
+	if obt == tableObject && len(localTempTableList) > 0 {
```

Comment:
This part is duplicate with `DDLExec.Next`, why?

Reply:
if all tables are local temp table, it could be dropped directly, but if tables are consists of local temp and normal tables, it should check whether table exists, then drop it